### DISCUSSION TITLE
Holograms

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#644] [Help] [answered]
Hi, you can spawn an hologram by doing this: "Hologram(instance, pos, component)". 
now I want to store this hologram in a variable by doing this: "var hologram = Hologram(instance, pos, component)."

now is my question: how can I spawn the hologram if I store them in a variable.
hologram.entity.spawn() doesn't work.

**Comment by TheMode** [comment]
Why would you want to store it in a field? You can cache the component independently if you really want to.

**Reply by DevBlokJochem** [reply]
because I want to edit/remove the hologram.

**Reply by TheMode** [reply]
You can already do that. But if you want more control over the hologram entity (e.g. for spawn), feel free to make your own.

**Reply by DevBlokJochem** [reply]
How can I get the hologram if it isn't a variable?

**Reply by TheMode** [reply]
I do not understand the question, what prevent you from putting the hologram in a field? The current requirement is just that the entity must spawn on initialization.

**Comment by emortaldev** [comment, marked as answer]
hologram = Hologram()

hologram.setInstance()
hologram.setCustomName or something like that, you can still edit it later or if you want to remove it you can do
hologram.remove()

Maybe use an entity if that doesn't work for some reason

**Reply by DevBlokJochem** [reply]
I was searching for a .spawn or something, but I forgot that that was called "setInstance"
